File: solutions/cpp/perfect-numbers/1/perfect_numbers.cpp
#include "perfect_numbers.h"

#include <cmath>
#include <vector>

namespace perfect_numbers {
enum_classify classify(int in1) {
  int tmp{};
  for (int i = 1; i <= in1 / 2; i++) {
    double x = (in1 / i) * 1.0;
    if (std::floor(x) == std::ceil(x)) {
      tmp += i;
    }
  }
  return tmp == in1 ? perfect : tmp < in1 ? deficient : abundant;
}
}  // namespace perfect_numbers

File: solutions/lua/perfect-numbers/1/perfect-numbers.lua
local function aliquot_sum(n)
    local result = 0
    for i = 1, n - 1, 1 do if n % i == 0 then result = result + i end end
    return result
end

local function classify(n)
    if aliquot_sum(n) < n then
        return 'deficient'
    elseif aliquot_sum(n) > n then
        return 'abundant'
    else
        return 'perfect'
    end
end

return {aliquot_sum = aliquot_sum, classify = classify}
